packages/html-to-slate-ast/src/index.ts

```diff
@@ -137,14 +142,51 @@ function deserialize(el: Node) {
       return jsx('element', attrs, listItemChildren);
     } else if (
       nodeName === 'TABLE' &&
-      !children.find((node: ChildNode) => node.nodeName === 'THEAD')
```

Comment:
This line was incorrect and did not prevent a `thead` from being inserted even if there was a `thead` in the input.